packages/sanity/src/structure/comments/plugin/field/CommentsField.tsx

```diff
@@ -76,6 +77,7 @@ function CommentFieldInner(props: FieldProps) {
   const {element: boundaryElement} = useBoundaryElement()
 
   const currentUser = useCurrentUser()
+  const commentsEnabled = useCommentsEnabled()
```

Comment:
It would be nice to avoid using this hook in here and only render this component when comments is enabled. I think making creating a new `CommentsFieldUpsell` component (see earlier comment) would be nice and separate the concerns of the enabled field and the upsell field logic.